cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -171,7 +171,11 @@ private String getSpecJson(Application app, final String bundleVersion, final St
         appConfig = (Config) configToken.getRawType().newInstance();
       } else {
         try {
-          appConfig = GSON.fromJson(configString, configToken.getType());
```

Comment:
Can you explain a bit? When is `configToken.getType()` returns if the application has extends without specifying the parameterized type (I guess that's what the fix is about)?